agent/cache-types/connect_ca_leaf_test.go

```diff
@@ -162,17 +163,28 @@ func TestConnectCALeaf_changingRoots(t *testing.T) {
 		QueryMeta: structs.QueryMeta{Index: 1},
 	}
 
+	// We need this later but needs to be defined so we sign second CSR with it
+	// otherwise we break the cert root checking.
+	caRoot2 := connect.TestCA(t, nil)
+
 	// Instrument ConnectCA.Sign to return signed cert
 	var resp *structs.IssuedCert
 	var idx uint64
+
 	rpc.On("RPC", "ConnectCA.Sign", mock.Anything, mock.Anything).Return(nil).
 		Run(func(args mock.Arguments) {
+			ca := caRoot
+			cIdx := atomic.AddUint64(&idx, 1)
+			if cIdx > 1 {
+				// Second time round use the new CA
+				ca = caRoot2
+			}
 			reply := args.Get(2).(*structs.IssuedCert)
-			leaf, _ := connect.TestLeaf(t, "web", caRoot)
+			leaf, _ := connect.TestLeaf(t, "web", ca)
 			reply.CertPEM = leaf
 			reply.ValidAfter = time.Now().Add(-1 * time.Hour)
 			reply.ValidBefore = time.Now().Add(11 * time.Hour)
-			reply.CreateIndex = atomic.AddUint64(&idx, 1)
+			reply.CreateIndex = cIdx
```

Comment:
Changes necessary because we used to keep signing with the first CA key before but that breaks blocking behaviour now as we detect the CA is different to expected in the final call. Now we have to model the change more accurately.